pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -470,20 +472,42 @@ func (pl *VolumeBinding) Score(ctx context.Context, cs *framework.CycleState, po
 	if !ok {
 		return 0, nil
 	}
-	// group by storage class
+
 	classResources := make(classResourceMap)
-	for _, staticBinding := range podVolumes.StaticBindings {
-		class := staticBinding.StorageClassName()
-		storageResource := staticBinding.StorageResource()
-		if _, ok := classResources[class]; !ok {
-			classResources[class] = &StorageResource{
-				Requested: 0,
-				Capacity:  0,
+	if len(podVolumes.StaticBindings) != 0 || !pl.fts.EnableStorageCapacityScoring {
+		// group static binding volumes by storage class
+		for _, staticBinding := range podVolumes.StaticBindings {
+			class := staticBinding.StorageClassName()
+			storageResource := staticBinding.StorageResource()
+			if _, ok := classResources[class]; !ok {
+				classResources[class] = &StorageResource{
+					Requested: 0,
+					Capacity:  0,
+				}
+			}
+			classResources[class].Requested += storageResource.Requested
+			classResources[class].Capacity += storageResource.Capacity
+		}
+	} else {
+		// group dynamic binding volumes by storage class
+		for _, provision := range podVolumes.DynamicProvisions {
+			if provision.Capacity == nil {
+				continue
+			}
+			class := *provision.PVC.Spec.StorageClassName
+			if _, ok := classResources[class]; !ok {
+				classResources[class] = &StorageResource{
+					Requested: 0,
+					Capacity:  provision.Capacity.Capacity.Value(),
+				}
+			} else {
+				classResources[class].Capacity = provision.Capacity.Capacity.Value()
 			}
```

Comment:
To make it consistent with StaticBindings code:
```suggestion
			if _, ok := classResources[class]; !ok {
				classResources[class] = &StorageResource{
					Requested: 0,
					Capacity:  0,
				}
			}
			classResources[class].Capacity = provision.Capacity.Capacity.Value()
```

Reply:
Btw, shouldn't it be +=?

Reply:
Yes, that's correct. I have addressed it.

Reply:
> Btw, shouldn't it be +=?

No, actually that's not correct. For example, if a Pod requests two 50GB volumes from a StorageClass with 100GB of capacity on a node, this part of the code will be executed twice. In that case, using `+=` would incorrectly set `classResources[class].Capacity` to 200GB.

Reply:
Okay, so please add a comment there. And the DynamicProvision.Capacity could be renamed to the NodeCapacity

Reply:
I have addressed it.